File: app/javascript/tailmix/runtime/operations/arithmetic.js
// Operations related to arithmetic.
export const ArithmeticOperations = {
    add: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 + val2;
    },

    subtract: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 - val2;
    },

    mod: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 % val2;
    },
};

File: lib/tailmix/scripting/operations/arithmetic.rb
# frozen_string_literal: true

module Tailmix
  module Scripting
    module Operations
      # Contains arithmetic operations.
      module Arithmetic
        extend self

        def add(interpreter, args)
          interpreter.eval(args[0]) + interpreter.eval(args[1])
        end

        def subtract(interpreter, args)
          interpreter.eval(args[0]) - interpreter.eval(args[1])
        end

        def mod(interpreter, args)
          interpreter.eval(args[0]) % interpreter.eval(args[1])
        end

        # We can add multiply, divide etc. here later
      end
    end
  end
end
